pkg/performanceprofile/cmd/render/render.go

```diff
@@ -143,22 +209,102 @@ func render(inputDir, outputDir string) error {
 		partitioningMode = &infra.Status.CPUPartitioning
 	}
 
-	if isLegacySNOWorkloadPinningMethod(mcConfigs, infra, partitioningMode) {
-		legacyAllNodes := apicfgv1.CPUPartitioningAllNodes
-		partitioningMode = &legacyAllNodes
-	}
-
-	if err := genBootstrapWorkloadPinningManifests(partitioningMode, outputDir, defaultMCPNames...); err != nil {
-		return err
-	}
+	if renderTune {
+		if len(tune) == 0 {
+			klog.Warning("zero tune patch were found")
+		}
+		if isLegacySNOWorkloadPinningMethod(mcConfigs, infra, partitioningMode) {
+			legacyAllNodes := apicfgv1.CPUPartitioningAllNodes
+			partitioningMode = &legacyAllNodes
+		}
 
-	// If the user supplies extra machine pools, we ingest them here
-	for _, pool := range mcPools {
-		if err := genBootstrapWorkloadPinningManifests(partitioningMode, outputDir, pool.Name); err != nil {
+		if err := genBootstrapWorkloadPinningManifests(partitioningMode, outputDir, defaultMCPNames...); err != nil {
 			return err
 		}
-	}
 
+		// If the user supplies extra machine pools, we ingest them here
+		for _, pool := range mcPools {
+			if err := genBootstrapWorkloadPinningManifests(partitioningMode, outputDir, pool.Name); err != nil {
```

Comment:
not really sure why this is needed.
Would not be enough to handle kernel arguments?

Reply:
Right, other mcps are not needed. Remained from another try...
Will remove